pkgs/development/python-modules/openai/default.nix

```diff
@@ -29,29 +40,28 @@ buildPythonPackage rec {
     owner = "openai";
     repo = "openai-python";
     rev = "v${version}";
-    hash = "sha256-bwv7lpdDYlk+y3KBjv7cSvaGr3v02riNCUfPFh6yv1I=";
+    hash = "sha256-dq0as07a5gSFbwLFjxKHBZFB51xjAPqs1FFrp0ScXBY=";
   };
 
   propagatedBuildInputs = [
-    numpy
-    openpyxl
-    pandas
-    pandas-stubs
+    aiohttp
     requests
     tqdm
+  ] ++ lib.optionals (pythonOlder "3.8") [
     typing-extensions
   ];
```

Comment:
We can add the optional dependencies here.

Reply:
Are you suggesting that we include all optional dependencies in `propagatedBuildInputs`?

I asked because in #194919 `passthru.optional-dependencies` was created. Before that change was made I used to just include all dependencies (optional or otherwise) in `propagatedBuildInputs`. I had assumed that change was made so that by default the packages would not include the optional dependencies, while giving folks an easy way to add the additional dependencies themselves if they wanted to include them. Which made sense to me.

I made the change to `openai` (defined in `all-packages.nix`) to include all the optional dependencies, so that folks using the Python package for the CLI utility would get all the functionality by default, while allowing folks who are using the Python package as a library to use `pythonXPackages.openai` and include only the optional dependencies they wanted/needed if any.

(Of course it may be that I'm confused or misunderstanding you, and therefore my comment doesn't make much sense. If that's the case, sorry about that.)

Reply:
> I asked because in https://github.com/NixOS/nixpkgs/pull/194919 passthru.optional-dependencies was created. Before that change was made I used to just include all dependencies (optional or otherwise) in propagatedBuildInputs. I had assumed that change was made so that by default the packages would not include the optional dependencies, while giving folks an easy way to add the additional dependencies themselves if they wanted to include them. Which made sense to me.

Yes, that's correct.



> I made the change to `openai` (defined in `all-packages.nix`) to include all the optional dependencies, so that folks using the Python package for the CLI utility would get all the functionality by default, while allowing folks who are using the Python package as a library to use `pythonXPackages.openai` and include only the optional dependencies they wanted/needed if any.

That wasn't obvious to me. I think in other packages we do a ``full`` variant. At least for me the current behavior would be a bit unexpected.

Reply:
Done.